app/views/users/_user.html.haml

```diff
@@ -13,26 +13,24 @@
           %span.badge= "Student"
       %div.action_menu_header_right
         %ul.menu
-          - if (user.changeable?(current_visitor))
-            - if (portal_teacher = user.portal_teacher) && policy(portal_teacher).show?
-              %li.menu=link_to 'Teacher Page', portal_teacher_path(portal_teacher)
-            - if (portal_student = user.portal_student) && policy(portal_student).show?
-              %li.menu=link_to 'Student Page', portal_student_path(portal_student)
-            - if user.has_role?("admin") && !current_visitor.has_role?("admin")
-              %li.menu=link_to 'edit', edit_by_project_admin_user_path(user)
-            - else
-              %li.menu=link_to 'edit', edit_user_path(user)
-            %li.menu=link_to 'Reset Password', reset_password_user_path(user)
-            - if user.state == 'pending'
-              %li.menu=link_to 'Activate', confirm_user_path(user)
-            - if(user != current_visitor)
-              - switch_user_form_id = "switch_user_id_#{user.id}"
-              %li.menu=link_to 'Switch', 'javascript: void(0);',:onclick => "$('#{switch_user_form_id}').submit();"
-              = form_for user, :url => switch_user_path(user), :html => { :method => :put,:class=> "hidden" ,:id=> switch_user_form_id}  do |form|
-                = hidden_field_tag 'user[id]', user.id
-                = hidden_field_tag 'commit', 'Switch'
-          - elsif current_visitor.is_project_admin? && user.portal_teacher
+          - if user.portal_teacher && policy(user).teacher_page?
+            %li.menu=link_to 'Teacher Page', portal_teacher_path(user.portal_teacher)
+          - if user.portal_student && policy(user).student_page?
+            %li.menu=link_to 'Student Page', portal_student_path(user.portal_student)
+          - if policy(user).edit_by_project_admin?
             %li.menu=link_to 'edit', edit_by_project_admin_user_path(user)
+          - if policy(user).edit_by_admin?
+            %li.menu=link_to 'edit', edit_user_path(user)
```

Comment:
Previously this was either one or the other situation. Will there be a case now where a user sees two edit links?